clippy_lints/src/methods.rs

```diff
@@ -490,6 +490,36 @@ declare_lint! {
     "using `.get().unwrap()` or `.get_mut().unwrap()` when using `[]` would work instead"
 }
 
+/// **What it does:** Checks for the use of `.extend(s.chars())` where s is a
+/// `&str` or `String`.
+///
+/// **Why is this bad?** `.push_str(s)` is clearer and faster
```

Comment:
Ok for clearer, but I'm surprised `extend` is not specialized to get the same speed